### PR TITLE
Fix contact api search

### PR DESCRIPTION
**Pull request by mplekunov**

### CODE REVIEW

**Comment by mplekunov** [comment]
I tested it, but I'm not sure if I covered every possible outcome, I would appreciate if you double check it on your local machine, just to be sure. @nashirj

**Comment by mplekunov** [comment]
Also, when we add contact, do you want to pass empty strings to the fields that are optional for initialization (phone, email, contactImage) ? 

There are 2 ways of doing this actually:
1. I will require optional fields to be in the body of the request (so like if your form data doesn't have fields filled I will return an error)
2. I will not require optional fields to be in the body of the request (if you have nothing to pass to the phone/email/contactImage - don't include them in the form-data)

**Comment by mplekunov** [comment]
> Also, when we add contact, do you want to pass empty strings to the fields that are optional for initialization (phone, email, contactImage) ?
> 
> There are 2 ways of doing this actually:
> 
> 1. I will require optional fields to be in the body of the request (so like if your form data doesn't have fields filled I will return an error)
> 2. I will not require optional fields to be in the body of the request (if you have nothing to pass to the phone/email/contactImage - don't include them in the form-data)

solved, option 1 was chosen by front-end

File: API/users/UpdateUser.php [hunk 1]
```diff
@@ -9,15 +9,24 @@
     
     require_once __DIR__ . '/../database/Database.php';
 
-    $payload = RequestReceiver::receivePOST();
-    if ($payload === false || !isset($payload['ID']))
+    $payload = RequestReceiver::recievePUT();
```
Comment (nashirj) [inline API/users/UpdateUser.php:12]: Typo, should be receive

File: API/contacts/DeleteContact.php [hunk 2]
```diff
@@ -14,6 +11,8 @@
     
     require_once __DIR__ . '/../database/Database.php';
 
+    require_once __DIR__ . '/../server/ServerException.php';
+
     $payload = RequestReceiver::recieveDELETE();
```
Comment (nashirj) [inline API/contacts/DeleteContact.php:16]: Typo, should be receive, pls find and fix all these if you can

**Review by nashirj — APPROVED** [review]

**Pull request merged**